File: Python/1-25/03.py
#!/usr/bin/python
import math
print(max(i for i in range(3, int(math.sqrt(600851475143)) + 2, 2) if 600851475143 % i == 0 and not any(i % j == 0 for j in range(3, int(math.sqrt(i))+2, 2))))

File: C/1-25/03.c
#include <stdio.h>
#include <math.h>

int prime_check(int n)
{
    int i;
    for (i = 3; i <= sqrt(n); i += 2) 
        if (n % i == 0) 
            return 0;
    return 1;
}

int main(int argc, char *argv[])
{
    long n = 600851475143;
    long i;
    for (i = sqrt(n) - 1; i > 3; i -= 2) 
        if (n % i == 0 && prime_check(i)) {
            printf("%d", i);
            break;
    }
    return 0;
}
